A72-open-telemetry-tracing.md

```diff
@@ -0,0 +1,428 @@
+propagators that users configure into gRPC OpenTelemetry, e.g. w3c.
+
+### Grpc OpenTelemetry Tracing API
+This section talks about enabling and configuring OpenTelemetry tracing.
+The OpenTelemetry API will coexist with the 
+OpenCensus API until the latter is dropped. Only one "grpc-trace-bin" header 
+will be sent for a single RPC.
+
+The APIs are different among languages due to different underlying infrastructures.
+
+#### Java
+In Java, it will be part of global interceptors, so that the interceptors are 
+managed in a more sustainable way and user-friendly. As a prerequisite, the stream
+tracer factory API will be stabilized. OpenTelemetryModule will be created with 
+an OpenTelemetryAPI instance passing in for necessary configurations. 
+Users can also rely on SDK autoconfig extension that configure the sdk object 
+through environmental variables or Java system properties, then obtain the sdk 
+object passed in to gRPC.
+
+```Java
+// Construct OpenTelemetry to be passed to gRPC OpenTelemetry module for 
+// trace and metrics configurations.
+SdkTracerProvider sdkTracerProvider = SdkTracerProvider
+  .builder()
+  .addSpanProcessor(
+     BatchSpanProcessor.builder(exporter).build())
+  .build();
+
+OpenTelemetry openTelemetry = OpenTelemetrySdk.builder()
+  .setTracerProvider(sdkTracerProvider)
+  .setMeterProvider(...)
+  .setPropagators(
+     ContextPropagators.create(GrpcTraceBinTextMapPropagator.getInstance()))
+  .build();
+
+// Alternatively, use auto configuration:
+// OpenTelemetry openTelemetry = 
+// AutoConfiguredOpenTelemetrySdk.getOpenTelemetrySdk().
+
+// OpenTelemetryModule.getInstance() will be using GlobalOpenTelemetry
+OpenTelemetryModule otModule = OpenTelemetryModule.getInstance(openTelemetry);
+
+GlobalInterceptors.setInterceptors(
+    Arrays.asList(otModule.getClientTracingInterceptor()),
+    Arrays.asList(otModule.getServerTracerFactory()));
+
+```
+
+#### C++
+In C++, it will be a method that mirrors OpenCensus API.
+
+TODO: update C++ API.
+
+```C++
+// Enable OpenTelemetry based tracing. Similar to
+// RegisterOpenCensusPlugin(). TracerProvider is configured via sdk separately.
+void RegisterOpenTelemetryTracingPlugin();
+
+```
+
+#### Go
+In Go, the OpenTelemetry stream tracers and interceptors will be provided for users to install.
+
+TODO: add Go API.
+
+### Migrate to OpenTelemetry: cross-process networking concerns
+When clients first introduce gRPC OpenTelemetry, for the time window when the 
+gRPC client and server have mixed plugins of OpenTelemetry and OpenCensus, 
+with `grpcTraceBinPropagator` users can do migration easily with the compatibility
+guaranteed. It is encouraged to use `grpcTraceBinPropagator` that propagates
+`grpc-trace-bin` header for migration because of the following advantages:
+* Simplified migration path, no migration phase deployments.
+* Binary header is more efficient.
+
+A binary formatter implementation for OpenTelemetry is needed in each language, 
```

Comment:
It isn't clear this is talking about grpc-trace-bin. You can call it a grpc-trace-bin formatter.